File: Top 100 Codes/02-Working with Numbers/12-Quadrants in which a given coordinate lies/Quadrantsinwhichagivencoordinatelies.cpp
#include <iostream>
using namespace std;

// Function to determine the quadrant of a point
void findQuadrant(int x, int y) {
    if (x > 0 && y > 0) {
        cout << "The point (" << x << ", " << y << ") lies in Quadrant I." << endl;
    } else if (x < 0 && y > 0) {
        cout << "The point (" << x << ", " << y << ") lies in Quadrant II." << endl;
    } else if (x < 0 && y < 0) {
        cout << "The point (" << x << ", " << y << ") lies in Quadrant III." << endl;
    } else if (x > 0 && y < 0) {
        cout << "The point (" << x << ", " << y << ") lies in Quadrant IV." << endl;
    } else if (x == 0 && y == 0) {
        cout << "The point (" << x << ", " << y << ") lies at the Origin." << endl;
    } else if (x == 0) {
        cout << "The point (" << x << ", " << y << ") lies on the Y-axis." << endl;
    } else if (y == 0) {
        cout << "The point (" << x << ", " << y << ") lies on the X-axis." << endl;
    }
}

int main() {
    int x, y;

    // Take user input for the coordinates
    cout << "Enter the X-coordinate: ";
    cin >> x;
    cout << "Enter the Y-coordinate: ";
    cin >> y;

    // Call the function to determine the quadrant
    findQuadrant(x, y);

    return 0; // Indicate successful execution
}

File: Top 100 Codes/02-Working with Numbers/12-Quadrants in which a given coordinate lies/Quadrantsinwhichagivencoordinatelies.py
# Take user input for coordinates and initialize variables
x, y = map(int, input("Enter the values for X and Y (separated by a space): ").split())

# Check if the point lies in the First quadrant
if x > 0 and y > 0:
    print(f"The point ({x}, {y}) lies in the First quadrant.")

# Check if the point lies in the Second quadrant
elif x < 0 and y > 0:
    print(f"The point ({x}, {y}) lies in the Second quadrant.")

# Check if the point lies in the Third quadrant
elif x < 0 and y < 0:
    print(f"The point ({x}, {y}) lies in the Third quadrant.")

# Check if the point lies in the Fourth quadrant
elif x > 0 and y < 0:
    print(f"The point ({x}, {y}) lies in the Fourth quadrant.")

# Check if the point lies at the origin
elif x == 0 and y == 0:
    print(f"The point ({x}, {y}) lies at the origin.")

# Check if the point lies on the X-axis
elif y == 0 and x != 0:
    print(f"The point ({x}, {y}) lies on the X-axis.")

# Check if the point lies on the Y-axis
elif x == 0 and y != 0:
    print(f"The point ({x}, {y}) lies on the Y-axis.")
